lib/sendgrid_helper.rb

```diff
@@ -1,18 +1,43 @@
 module SendgridHelper
   def self.spam_reported?(email)
-    handle_response(spam_reported_url(email))
+    handle_response(spam_reported_url(email)) if can_access_sendgrid?
   end
 
   def self.bounced?(email)
-    handle_response(bounced_url(email))
+    handle_response(bounced_url(email)) if can_access_sendgrid?
+  end
+
+  def self.smtp_alive?(host, port)
+    Net::SMTP.start(host, port) do |smtp|
+      smtp.enable_starttls_auto
+      smtp.ehlo(Socket.gethostname)
+      smtp.finish
+    end
+    true
+  rescue StandardError => e
+    false
+  end
+
+  private
+
+  def self.can_access_sendgrid?
+    ENV['SMTP_USERNAME'] and ENV['SMTP_PASSWORD']
```

Comment:
Looking at this again, we should probably use the existing config variables, rather than directly accessing environment variables. I *think* you should be able to get them using `config.action_mailer.smtp_settings.user_name`?